app/models/miq_group.rb

```diff
@@ -18,17 +22,20 @@ class MiqGroup < ActiveRecord::Base
 
   delegate :self_service?, :limited_self_service?, :to => :miq_user_role, :allow_nil => true
 
-  validates_presence_of   :description, :guid
-  validates_uniqueness_of :description, :guid
+  validates :description, :guid, :presence => true, :uniqueness => true
+  validate :validate_default_tenant, :on => :update, :if => :tenant_id_changed?
 
   before_destroy do |g|
     raise "Still has users assigned." unless g.users.empty?
-    raise "A read only group cannot be deleted." if g.read_only
+    raise "A tenant default group can not be deleted" if g.tenant_group? && g.referenced_by_tenant?
```

Comment:
Yeah, Keenan actually changed that line at the sam time or shortly after I added the comment